rllib/agents/trainer_config.py

```diff
@@ -0,0 +1,1015 @@
+import copy
+import gym
+from typing import (
+    Any,
+    Callable,
+    Optional,
+    Union,
+)
+
+from ray.rllib.agents.callbacks import DefaultCallbacks
+from ray.rllib.env.env_context import EnvContext
+from ray.rllib.models import MODEL_DEFAULTS
+from ray.rllib.utils.typing import (
+    EnvConfigDict,
+    EnvType,
+    PartialTrainerConfigDict,
+    ResultDict,
+    TrainerConfigDict,
+)
+from ray.tune.logger import Logger
+
+
+class TrainerConfig:
+    """A RLlib TrainerConfig builds an RLlib trainer from a given configuration.
+
+    Example:
+       >>> from ray.rllib.agents.callbacks import MemoryTrackingCallbacks
+       >>> config = TrainerConfig.training(gamma=0.9, lr=0.01)
+       ...              .environment(env="CartPole-v1")
+       ...              .resources(num_gpus=0)
+       ...              .rollouts(num_rollout_workers=4)
```

Comment:
maybe mention that subclasses implementing this simply call `build` to retrieve the actual trainer?

Reply:
done